cors.go

```diff
@@ -10,16 +10,20 @@ import (
 type CORSOption func(*cors) error
 
 type cors struct {
-	h                http.Handler
-	allowedHeaders   []string
-	allowedMethods   []string
-	allowedOrigins   []string
-	exposedHeaders   []string
-	maxAge           int
-	ignoreOptions    bool
-	allowCredentials bool
+	h                      http.Handler
+	allowedHeaders         []string
+	allowedMethods         []string
+	allowedOrigins         []string
+	allowedOriginValidator originValidator
+	exposedHeaders         []string
+	maxAge                 int
+	ignoreOptions          bool
+	allowCredentials       bool
 }
 
+// originValidator takes an origin string and returns whether or not that origin is allowed.
+type originValidator func(string) bool
```

Comment:
This should be exported so that `AllowedOriginValidator`'s argument type is in the API docs.